backend-es/test/snapshots-out/Snapshot.CaseGuarded.js

```diff
@@ -73,18 +50,8 @@ const test2 = v => {
 };
 const test1 = v => {
   if (v < 1) { return "n: " + Data$dShow.showIntImpl(v); }
-  if (v > 1) {
-    if (v < 100) { return "1 < x < 100: " + Data$dShow.showIntImpl(v); }
-    if (v > 1) {
-      if (v < 50) { return "1 < x < 50: " + Data$dShow.showIntImpl(v); }
-      return "catch";
-    }
-    return "catch";
-  }
-  if (v > 1) {
-    if (v < 50) { return "1 < x < 50: " + Data$dShow.showIntImpl(v); }
-    return "catch";
-  }
+  if (v > 1 && v < 100) { return "1 < x < 100: " + Data$dShow.showIntImpl(v); }
+  if (v > 1 && v < 50) { return "1 < x < 50: " + Data$dShow.showIntImpl(v); }
```

Comment:
This is correct, but the duplicate `v > 1` is another optimization we could check for. Moreover, if `v < 50`, we also know that `v < 100`.

Reply:
That's what I thought at first, but this is a result of the duplicate tests in pattern guard desugaring.